### DISCUSSION TITLE
Blocked ports when changing directories to publish website

### DISCUSSION THREAD

**Discussion by ghost** [community/community#60428] [Programming Help]
I just changed the directory and the document root of my xammp config, because i want to change web to direct to another folder, but after changing it, i cant restart the xammp control panel as it says that my port 80 and port 443 has been used and have a crashed in other services, i have tried changing the listen 80 to another port but does not seem to work too.

![image](https://github.com/community/community/assets/104344742/6ff0c3af-e7ea-4a98-99f1-701ebdb5b4a0)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @whywhyyyyy ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by sbarissonmez** [comment]
Hi @whywhyyyyy 

If you have changed the directory and document root of your XAMPP configuration and are facing issues with ports 80 and 443 being used or services crashing, here are some steps you can take to troubleshoot and resolve the problem:

1. Check for Other Applications or Services Using Ports 80 and 443:
   - Ports 80 and 443 are commonly used by web servers like Apache (which is included in XAMPP) and other applications like Skype, IIS, or other web servers running on your system.
   - Ensure that no other applications or services are already using ports 80 and 443. You can use the following command in the command prompt or terminal to check:

     ```
     netstat -ano | findstr :80
     netstat -ano | findstr :443
     ```

   - The above commands will show if any processes are currently using ports 80 and 443. Note down the Process ID (PID) of the processes and consider stopping or reconfiguring them to free up the ports.

2. Check Apache Configuration Files:
   - Verify that you have updated the correct Apache configuration file after changing the directory and document root. The main Apache configuration file is usually named `httpd.conf` or `apache2.conf`.
   - Look for occurrences of `Listen 80` and `Listen 443` in the configuration file. If you changed the ports in the configuration, make sure that the new ports are reflected correctly.

3. Ensure Proper Syntax in Apache Configuration:
   - Ensure that there are no syntax errors in the Apache configuration file after making changes. An error in the configuration could prevent Apache from starting.

4. Restart XAMPP Services:
   - After making any changes to the configuration files, restart the XAMPP services to apply the changes. You can do this from the XAMPP Control Panel.

5. Check Windows Firewall or Antivirus Settings:
   - Sometimes, Windows Firewall or antivirus software can interfere with port usage. Check their settings to ensure that they are not blocking ports 80 and 443.

6. Run XAMPP Control Panel as Administrator:
   - Try running the XAMPP Control Panel as an administrator. Right-click on the XAMPP Control Panel shortcut and select "Run as administrator." This may resolve some permission-related issues.

7. Reboot the System:
   - If all else fails, try rebooting your system. This can help free up any resources and resolve conflicts.

After taking these steps, try starting XAMPP again and check if the issues with ports 80 and 443 are resolved. If you are still encountering problems, review any error messages or logs to identify the cause of the issue. Additionally, consider seeking help from the XAMPP community or forums, as they may have encountered similar issues and can provide specific guidance.
